db/db_impl/db_impl.h

```diff
@@ -113,6 +113,54 @@ class Directories {
   std::unique_ptr<FSDirectory> wal_dir_;
 };
 
+// Interface to block and signal the DB in case of stalling writes by
+// WriteBufferManager. Each DBImpl object contains ptr to WBMStallInterface.
+// When DB needs to be blocked or signalled by WriteBufferManager,
+// state_for_wbm_ state is changed accordingly.
+class WBMStallInterface {
```

Comment:
Can this be defined as a trait in ```write_buffer_manager.h``` and ```DBImpl``` implements it? Or define an abstract class in ```write_buffer_manager.h``` and derive ```WBMStallInterface``` from it? That would avoid ```static_cast```ing it.